### PR TITLE
Support spliting mod deps.

### LINKED ISSUE #667: Split dependencies when using split source sets
Right now compile time sided safety is only provided for Minecraft, do the same for 3rd party libraries.

### PR DESCRIPTION
**Pull request by modmuss50**

### CODE REVIEW

File: src/main/java/net/fabricmc/loom/util/SourceRemapper.java [hunk 1]
```diff
@@ -61,13 +61,7 @@ public SourceRemapper(Project project, boolean toNamed) {
...
 	}
 
-	public static void remapSources(Project project, File input, File output, boolean named, boolean reproducibleFileOrder, boolean preserveFileTimestamps) {
-		SourceRemapper sourceRemapper = new SourceRemapper(project, named);
-		sourceRemapper.scheduleRemapSources(input, output, reproducibleFileOrder, preserveFileTimestamps);
-		sourceRemapper.remapAll();
-	}
-
-	public void scheduleRemapSources(File source, File destination, boolean reproducibleFileOrder, boolean preserveFileTimestamps) {
+	public void scheduleRemapSources(File source, File destination, boolean reproducibleFileOrder, boolean preserveFileTimestamps, Runnable compilationCallback) {
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/util/SourceRemapper.java]: Is the runnable ever used? There don't seem to be any changes doing that

Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/util/SourceRemapper.java]: Good spot, forgot to invoke it 👍

Comment (modmuss50) [reply src/main/java/net/fabricmc/loom/util/SourceRemapper.java]: and its named wrong

File: src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java [hunk 2]
```diff
@@ -168,49 +116,49 @@ private void remapJars(List<ModDependencyInfo> remapList) throws IOException {
...
+						project.getLogger().debug("Remapping access widener in {}", dependency.getInputFile());
+						byte[] remappedAw = AccessWidenerUtils.remapAccessWidener(accessWidenerData.content(), remapper.getEnvironment().getRemapper());
+						accessWidenerMap.put(dependency, new Pair<>(remappedAw, accessWidenerData.path()));
 					}
 
-					remapper.apply(outputConsumer, tagMap.get(info));
+					remapper.apply(outputConsumer, tagMap.get(dependency));
 				} catch (Exception e) {
-					throw new RuntimeException("Failed to remap: " + info.getRemappedNotation(), e);
+					throw new RuntimeException("Failed to remap: " + dependency, e);
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/configuration/mods/ModProcessor.java:164]: Might be good to add a toString to ModDependency to help debugging

**Comment by modmuss50** [comment]
This is feature complete and manually tested as working. It helped find a possible bug in TR 🎊 https://github.com/TechReborn/TechReborn/commit/88d448c3a57d56bc46828a422aede0d7b8e6dd12

**Pull request merged**